### DISCUSSION TITLE
Convert PyTorch loss to torchmetric Metric?

### DISCUSSION THREAD

**Discussion by charlesbmi** [Lightning-AI/torchmetrics#670] [Q&A]
I'm interested in using a `Metric` like [Poisson negative log likelihood loss](https://pytorch.org/docs/stable/generated/torch.nn.PoissonNLLLoss.html), which is already implemented in PyTorch as a `Module`.

Is there a helper-class to wrap existing PyTorch losses in a torchmetric `Metric`?

In particular, I would like to store and log the best epoch-accumulated loss so far (similar to this example with `Accuracy`: https://github.com/ashleve/lightning-hydra-template/blob/main/src/models/mnist_model.py#L93-L96). I looked at the `MeanSquaredError` source code as an example and know that I can re-implement a `Metric`, but seems like it could be nice to be able to easily wrap a PyTorch loss in a `Metric` class. Thanks!
